libs/index.js

```diff
@@ -92,24 +97,35 @@ function gPublish(options) {
 
     var bucket = storage.bucket(options.bucket);
 
-    var gcPah = normalizePath(options.base, file);
-
-    var gcFile = bucket.file(gcPah);
-
-    file.pipe(gcFile.createWriteStream({metadata: metadata}))
-        .on('error', done)
-        .on('finish', function() {
-          if (options.public) {
-            return gcFile.makePublic(function(err) {
-              logSuccess(gcPah);
-              done(err, file);
+    // get the filename
+    var fileToUpload = file.history[0];
```

Comment:
Can you explain me why you used `file.history` instead of `file.path`?
It's been a while since I've worked with vinyl files